packages/fast-components-styles-msft/src/utilities/density.ts

```diff
@@ -1,6 +1,146 @@
+export function heightNumber(
+    lines: number,
+    minusBorder: number = 0
+): (config: DesignSystem) => number {
+    return (config: DesignSystem): number => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier + designSystem.density) *
+                designSystem.designUnit *
+                lines -
+            minusBorder;
+        return value;
+    };
+}
+
+export function minHeight(
+    lines: number = 1,
+    minusBorder: number = 0
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier - 3) * designSystem.designUnit * lines -
+            minusBorder;
+        return toPx(value);
+    };
+}
+
+export function maxHeight(
+    lines: number = 1,
+    minusBorder: number = 0
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier + 3) * designSystem.designUnit * lines -
+            minusBorder;
+        return toPx(value);
+    };
+}
+
+/**
+ * Returns the standard horizontal padding for text and icons formatted in the provided unit or px by default.
+ *
+ * @param minusBorder Any border that should be removed from the overall content padding.
+ * @param unit The unit of measurement; px by default.
+ */
+export function padding(
+    minusBorder: number = 0,
+    unit?: string
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const offset: number = densityToTypeOffset(designSystem);
+        const value: number = paddingNumber(minusBorder)(designSystem);
+        return typeof unit === "string" ? `${value}${unit}` : toPx(value);
+    };
+}
+
+/**
+ * Returns the standard horizontal padding for text and icons as a number.
+ *
+ * @param minusBorder Any border that should be removed from the overall content padding.
+ * @param unit The unit of measurement; px by default.
+ */
+export function paddingNumber(minusBorder: number = 0): (config: DesignSystem) => number {
+    return (config: DesignSystem): number => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const offset: number = densityToTypeOffset(designSystem);
+        const value: number =
+            (designSystem.defaultPaddingMultiplier + offset) * designSystem.designUnit -
+            minusBorder;
+        return value;
+    };
+}
+
+/**
+ * Adds the font size and line height attributes for the type ramp according to the design system config.
+ *
+ * @param config The design system config.
+ */
+export function applyFontSize(
```

Comment:
I kind of think this could be more generally usefull. Really what we're doing is scaling typography with density - I think a more intuitive API might be to pass in a "default" typeramp ID that, with no density modifier, would be exaclty that type ID. but, given a density, it would scale accordingly.

Passing an "offset" requires you to know what the function internals are doing - to achieve a scaled "t5" I would need to do:  `applyFontSize(designSystem, -2)` as opposed to just sying `scaleTypeRamp("t5")(designSystem)`

Reply:
I was thinking in terms of offsets because I think it actually requires you to know less. Your t5 example is only because you happen to know that t7 is the default. I've added a design system setting for the default, so we could conceivably change it to 8, in which case a -2 offset would produce t6 text. So default exists, it's just centralized.

Reply:
I agree with Nick in terms of supporting more of a more intuitive API. Right now this seems like a tightly coupled black box

Reply:
Modified per our conversation.